frontend/src/metabase/query_builder/components/view/ViewFooter/LeftViewFooterButtonGroup.tsx

```diff
@@ -1,58 +1,136 @@
+  const isShowingRawTable = useSelector(getIsShowingRawTable);
+  const vizIcon = getIconForVisualizationType(question.display());
+  const isNative = question && Lib.queryDisplayInfo(question.query()).isNative;
+
+  useUnmount(() => {
+    // reset showing raw table, so new mount will default to viz
+    dispatch(setUIControls({ isShowingRawTable: false }));
+  });
 
   const dispatch = useDispatch();
+  const isVisualized = useSelector(getIsVisualized);
+
+  const data = useMemo(
+    () =>
+      [
+        isNative
+          ? null
+          : {
+              value: "editor",
+              label: (
+                <Tooltip label={t`Editor`}>
+                  <Icon
+                    name="notebook"
+                    onClick={() => {
+                      dispatch(setQueryBuilderMode("notebook"));
+                    }}
+                  />
+                </Tooltip>
+              ),
+            },
+        {
+          value: "table",
+          label: (
+            <Tooltip label={t`Results`}>
+              <Icon
+                aria-label={t`Switch to data`}
+                name="table2"
+                onClick={() => {
+                  dispatch(setUIControls({ isShowingRawTable: true }));
+                }}
+              />
+            </Tooltip>
+          ),
+        },
+        {
+          value: "visualization",
+          // TODO: also we need to add a spinner :boom:
+          label: (
+            <Tooltip label={t`Visualization`}>
+              <Icon
+                aria-label={t`Switch to visualization`}
+                name={vizIcon}
+                onClick={() => {
+                  dispatch(setUIControls({ isShowingRawTable: false }));
+                }}
+              />
+            </Tooltip>
+          ),
+        },
+      ].filter(isNotNull),
+    [dispatch, isNative, vizIcon],
+  );
 
   return (
-    <Group className={CS.flex1}>
-      <FooterButtonGroup>
-        <ViewButton
-          medium
-          labelBreakpoint="sm"
-          data-testid="viz-type-button"
-          active={isShowingChartTypeSidebar}
-          onClick={
-            isShowingChartTypeSidebar
-              ? () => dispatch(onCloseChartType())
-              : () => dispatch(onOpenChartType())
-          }
-        >
-          {t`Visualization type`}
-        </ViewButton>
-        <ViewButton
-          active={isShowingChartSettingsSidebar}
-          icon="gear"
-          iconSize={16}
-          medium
-          iconWithText
-          labelBreakpoint="sm"
+    <Flex gap="0.75rem">
+      {isVisualized && (
+        <EditorViewControl
+          value={isShowingRawTable ? "table" : "visualization"}
+          data={data}
+        />
+      )}
+      {(!isShowingRawTable || isVisualized) && (
+        <Button
+          radius="xl"
+          variant={isShowingChartSettingsSidebar ? "filled" : "default"}
+          styles={{
+            ...(!isShowingChartSettingsSidebar && {
+              root: {
```

Comment:
:warning:  We should not customize components like that. Can we use an existing variant, or should we introduce a new one?

Reply:
yes, we should have a new button 32px height

https://metaboat.slack.com/archives/C057WD5L0JG/p1729181236955289

Reply:
I meant color customization here

Reply:
we used to use the old button for such styling. I'll double check it with @kdoh

Reply:
@kamilmielnik [we're in transition phase](https://metaboat.slack.com/archives/C0645JP1W81/p1731590539423719?thread_ts=1731587774.888749&cid=C0645JP1W81), so later on we'll have such variant in the design system

so let's keep it as it is